jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/ServletPathSpec.java

```diff
@@ -270,6 +284,44 @@ private boolean isWildcardMatch(String path)
         return false;
     }
 
+    @Override
+    public MatchedPath matched(String path)
+    {
+        switch (_group)
+        {
+            case EXACT:
+                if (_declaration.equals(path))
+                    return new ServletMatchedPath(path, null); // TODO: return final matchedpath
+                break;
+            case PREFIX_GLOB:
+                if (isWildcardMatch(path))
+                {
+                    String pathMatch = path;
+                    String pathInfo = null;
+                    if (path.length() != (_specLength - 2))
+                    {
+                        pathMatch = path.substring(0, _specLength - 2);
+                        pathInfo = path.substring(_specLength - 2);
+                    }
+                    return new ServletMatchedPath(pathMatch, pathInfo);
+                }
+                break;
+            case SUFFIX_GLOB:
+                if (path.regionMatches((path.length() - _specLength) + 1, _declaration, 1, _specLength - 1))
+                    return new ServletMatchedPath(path, null);
+                break;
+            case ROOT:
+                // Only "/" matches
+                if ("/".equals(path))
+                    return new ServletMatchedPath("", path); // TODO: review this
+                break;
+            case DEFAULT:
+                // If we reached this point, then everything matches
+                return new ServletMatchedPath(path, null);
```

Comment:
this should also be a return of a known `MatchedPath`:
```suggestion
                return _matchedPath;
```